infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -334,37 +356,48 @@ public void multiRead(CorfuPayloadMsg<MultipleReadRequest> msg, ChannelHandlerCo
         }
     }
 
-    @ServerHandler(type = CorfuMsgType.INSPECT_ADDRESSES_REQUEST)
-    public void inspectAddresses(CorfuPayloadMsg<InspectAddressesRequest> msg,
-                                 ChannelHandlerContext ctx, IServerRouter r) {
-        log.trace("inspectAddresses: {}", msg.getPayload().getAddresses());
-        InspectAddressesResponse inspectResponse = new InspectAddressesResponse();
+    @ServerHandler(type = CorfuMsgType.LOG_RECOVERY_STATES_REQUEST)
+    public void readRecoveryStates(CorfuPayloadMsg<MultipleReadRequest> msg,
+                                   ChannelHandlerContext ctx, IServerRouter r) {
+        log.trace("readRecoveryStates: {}", msg.getPayload().getAddresses());
+
+        // Read garbage entries first to avoid losing garbage information since
```

Comment:
In your comment, you said that we should read garbage entry first. However, in the actual implementation, the logData is read first.

Reply:
Fixed.